cpp/src/Slice/Scanner.l

```diff
@@ -315,47 +316,59 @@ floating_literal    (({fractional_constant}{exponent_part}?)|((\+|-)?{dec}+{expo
 
   /* ========== Comments ========== */
 
-  /* Matches and records a triple-slash style doc comment. */
+  /* Matches and records a triple-slash style doc-comment. */
 <*>"///".* {
     currentUnit->addToDocComment(yytext + 3);
 }
 
   /* Matches and consumes a C++ style comment. */
 <*>"//".* {}
 
+  /* Matches the start of a C style doc-comment, and switches the scanner to the C_DOC_COMMENT state.
+   * Specifically we match the literal "/**" when it is _not_ followed by either a '/' or '*' character. */
+<*>"/**"/[^/*] {
+    yy_push_state(C_DOC_COMMENT);
+}
+
   /* Matches the start of a C style comment, and switches the scanner to the C_COMMENT state. */
 <*>"/*" {
     yy_push_state(C_COMMENT);
 }
 
   /* Matches any character except for newlines and adds them to the comments. '*' are matched one at a time to ensure
    * Flex scans '* /' correctly. Flex prioritizes longer matches over shorter ones, so '* /' will match before '*'. */
-<C_COMMENT>"*"     |
-<C_COMMENT>[^\n*]+ {
+<C_COMMENT,C_DOC_COMMENT>"*"     |
+<C_COMMENT,C_DOC_COMMENT>[^\n*]+ {
```

Comment:
In Flex, anything in angle brackets means: "only match this rule if you are in one of these start conditions".
We re-use these rules for both the `C_COMMENT` and `C_DOC_COMMENT` conditions, since they only difference is whether the text counts towards a doc-comment or not.